src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp

```diff
@@ -0,0 +1,386 @@
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "gate_setting_sun.h"
+
+enum Spells
+{
+    // Commander Rimok
+    SPELL_VISCOUS_FLUID_SUMMON      = 107078,
+    SPELL_VISCOUS_FLUID_DMG_UP      = 107091,
+    SPELL_VISCOUS_FLUID_DMG_DOWN    = 107122,
+    SPELL_FRENZIED_ASSAULT          = 107120,
+
+    // Add Generator
+    SPELL_PERIODIC_SPAWN_SWARMER    = 115052,
+    SPELL_PERIODIC_SPAWN_SABOTEUR   = 116621,
+
+    // Saboteur
+    SPELL_BOMBARD                   = 120559
+};
+
+enum Yells
+{
+    TALK_AGGRO,
+    TALK_DEATH,
+    TALK_INTRO,
+    TALK_SLAY,
+};
+
+enum Events
+{
+    EVENT_FRENZIED_ASSAULT  = 1,
+    EVENT_VISCOUS_FLUID     = 2
+};
+
+class boss_commander_rimok : public CreatureScript
+{
+    public:
+        boss_commander_rimok() : CreatureScript("boss_commander_rimok") {}
+
+        struct boss_commander_rimokAI : public BossAI
+        {
+            boss_commander_rimokAI(Creature* creature) : BossAI(creature, DATA_RIMOK)
+            {
+            }
+
+            void Reset()
+            {
+                _Reset();
+
+                events.ScheduleEvent(EVENT_FRENZIED_ASSAULT, urand(5000, 10000));
+                events.ScheduleEvent(EVENT_VISCOUS_FLUID,    urand(10000, 15000));
+            }
+
+            void EnterCombat(Unit* /*who*/)
+            {
+                _EnterCombat();
+                Talk(TALK_AGGRO);
+            }
+
+            void JustReachedHome()
+            {
+                instance->SetBossState(DATA_RIMOK, FAIL);
+                summons.DespawnAll();
+            }
+
+            void KilledUnit(Unit* /*u*/)
+            {
+                if (!urand(0, 1))
+                    Talk(TALK_SLAY);
+            }
+
+            void JustSummoned(Creature* /*summoned*/)
+            {
+            }
+
+            void UpdateAI(uint32 diff)
+            {
+                if (!UpdateVictim())
+                    return;
+
+                events.Update(diff);
+
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                switch (events.ExecuteEvent())
+                {
+                    case EVENT_FRENZIED_ASSAULT:
+                        DoCastVictim(SPELL_FRENZIED_ASSAULT);
+                        events.ScheduleEvent(EVENT_FRENZIED_ASSAULT, urand(10000, 15000));
+                        break;
+                    case EVENT_VISCOUS_FLUID:
+                        me->CastSpell(me, 107077, true);

```

Comment:
```cpp
                    case EVENT_VISCOUS_FLUID:
                        DoCast(me, SPELL_VISCOUS_FLUID_SUMMON, true);
```
Viscous Fluid enum changes:
```cpp
enum Spells
{
    // Commander Rimok
    SPELL_VISCOUS_FLUID_SUMMON      = 107077, // Effect: Summon (Viscos Fluid Stalker)
    SPELL_VISCOUS_FLUID_APPLY_AURA  = 107078, // Effect: Apply Aura: Periodically trigger spell
    SPELL_VISCOUS_FLUID_DMG_UP      = 107091, // Effect: Apply Aura: Mod Damage Done (Value: +10%  +20%)
    SPELL_VISCOUS_FLUID_DMG_DOWN    = 107122, // Effect: Apply Aura: Mod Damage Done (Value: -10%  -20%)
    SPELL_FRENZIED_ASSAULT          = 107120,

    // Add Generator
    SPELL_PERIODIC_SPAWN_SWARMER    = 115052,
    SPELL_PERIODIC_SPAWN_SABOTEUR   = 116621,

    // Saboteur
    SPELL_BOMBARD                   = 120559
};
```
Sorry for changing your enum names, I based them on the wowhead.com description:
- http://www.wowhead.com/spell=107077/viscous-fluid (Effect: Summon (Viscos Fluid Stalker))
- http://www.wowhead.com/spell=107078/viscous-fluid (Apply Aura: Periodically trigger spell)